src/EnergyPlus/EconomicLifeCycleCost.cc

```diff
@@ -389,41 +393,20 @@ void GetInputLifeCycleCostParameters(EnergyPlusData &state)
         //      \key StraightLine-40year
         //      \key None
         //      \default None
-        if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-3year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS3;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-5year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS5;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-7year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS7;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-10year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS10;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-15year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS15;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "ModifiedAcceleratedCostRecoverySystem-20year")) {
-            elcc->depreciationMethod = iDeprMethod::MACRS20;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "StraightLine-27year")) {
-            elcc->depreciationMethod = iDeprMethod::Straight27;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "StraightLine-31year")) {
-            elcc->depreciationMethod = iDeprMethod::Straight31;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "StraightLine-39year")) {
-            elcc->depreciationMethod = iDeprMethod::Straight39;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "StraightLine-40year")) {
-            elcc->depreciationMethod = iDeprMethod::Straight40;
-        } else if (UtilityRoutines::SameString(AlphaArray(6), "None")) {
-            elcc->depreciationMethod = iDeprMethod::None;
-        } else if (state.dataIPShortCut->lAlphaFieldBlanks(6)) {
-            elcc->depreciationMethod = iDeprMethod::None;
-            ShowWarningError(
-                state, CurrentModuleObject + ": The input field " + state.dataIPShortCut->cAlphaFieldNames(6) + "is blank. \"None\" will be used.");
-        } else {
-            elcc->depreciationMethod = iDeprMethod::None;
-            ShowWarningError(state,
-                             CurrentModuleObject + ": Invalid " + state.dataIPShortCut->cAlphaFieldNames(6) + "=\"" + AlphaArray(6) +
-                                 R"(". "None" will be used.)");
+        elcc->depreciationMethod = static_cast<DeprMethod>(getEnumerationValue(DeprMethodNamesUC, UtilityRoutines::MakeUPPERCase(AlphaArray(6))));
```

Comment:
@mjwitte maybe you can remind me here...can we not count on the input being all caps at this point?  Assuming we stick with a case-insensitive input format, even for epJSON values, we should just make sure the input processor upper cases everything, even the epJSON blob.  (Of course not the fields with `\retaincase`).  

It would be really nice to just know for sure that the AlphaArray was already upper case then we could eliminate so many MakeUPPERCase function calls and clean up so many chunks of code.

Reply:
At present, there's no global pass to push the input strings to UPPERcase. `getObjectItem` and `getAlphaFieldValue` return UPPERcase strings. Direct access of the JSON data does not.

Reply:
I don't think that is strictly necessary. There is nothing wrong with `makeUpperCase` as long as you only do it once per string. The most important thing is to avoid case-insensitive comparisons and there are other ways to do that. See [this issue](https://github.com/NREL/EnergyPlus/issues/9183).

Reply:
That's what I was afraid of, but I also see a good possibility there.  If the IP just makes one quick pass to upper case all the (non-retain-case) fields in the IDF or epJSON, then we would get rid of a huge amount of these `MakeUPPERCase` calls spread out all over the code.  And also it would make it perfectly clear to the developers that the data **will** be upper case as soon as the input processor is done with it.  No ambiguity that leads to extraneous string operations during input processing because the dev isn't sure if it will be upper case or not.  

I think this would be a great discussion to have, not here, but soon.  Thanks!

Reply:
Thanks @amirroth, I think I am in full agreement that there is nothing wrong with MakeUPPERCase, as long as it is only done once, but the problem is that we aren't quite sure if it's been done or not, so it ends up being done extra times.  If the input processor just made it upper case one time then we could be absolutely sure it's upper case and eliminate many MakeUPPERCase calls spread out of the code.  Anyway, nothing for this PR.  Thanks!

Reply:
I agree. If `MakeUpperCase` is done in input processor by default that's even better, but what I was saying is that we can eliminate a lot of the expensive case-insensitive comparisons even with no changes to the input processor or JSON module, because the problem is not `MakeUpperCase` inherently, it's the fact that SameString essentially does a `MakeUpperCase` on both inputs and it is called by many search routines.